libs/langchain/langchain/vectorstores/pinecone.py

```diff
@@ -342,54 +404,13 @@ def from_texts(
                     index_name="langchain-demo"
                 )
         """
-        try:
-            import pinecone
-        except ImportError:
-            raise ValueError(
-                "Could not import pinecone python package. "
-                "Please install it with `pip install pinecone-client`."
-            )
-
-        indexes = pinecone.list_indexes()  # checks if provided index exists
-
-        if index_name in indexes:
-            index = pinecone.Index(index_name)
-        elif len(indexes) == 0:
-            raise ValueError(
-                "No active indexes found in your Pinecone project, "
-                "are you sure you're using the right API key and environment?"
-            )
-        else:
-            raise ValueError(
-                f"Index '{index_name}' not found in your Pinecone project. "
-                f"Did you mean one of the following indexes: {', '.join(indexes)}"
-            )
-
-        for i in range(0, len(texts), batch_size):
-            # set end position of batch
-            i_end = min(i + batch_size, len(texts))
-            # get batch of texts and ids
-            lines_batch = texts[i:i_end]
-            # create ids if not provided
-            if ids:
-                ids_batch = ids[i:i_end]
-            else:
-                ids_batch = [str(uuid.uuid4()) for n in range(i, i_end)]
-            # create embeddings
-            embeds = embedding.embed_documents(lines_batch)
-            # prep metadata and upsert batch
-            if metadatas:
-                metadata = metadatas[i:i_end]
-            else:
-                metadata = [{} for _ in range(i, i_end)]
-            for j, line in enumerate(lines_batch):
-                metadata[j][text_key] = line
-            to_upsert = zip(ids_batch, embeds, metadata)
-
-            # upsert to Pinecone
-            _upsert_kwargs = upsert_kwargs or {}
-            index.upsert(vectors=list(to_upsert), namespace=namespace, **_upsert_kwargs)
-        return cls(index, embedding, text_key, namespace, **kwargs)
+        pinecone = cls.get_pinecone(
```

Comment:
I think looking at this line makes me feel more strongly we should probably rename `get_pinecone` to something more explicit - otherwise it's hard to quickly understand what this is doing when scanning the code.